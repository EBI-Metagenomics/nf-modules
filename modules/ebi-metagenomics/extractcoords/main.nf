
process EXTRACTCOORDS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:1.0.1--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:1.0.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(easel_coords_fasta)
    tuple val(meta2), path(matched_seqs_with_coords)

    output:
    tuple val(meta), path("sequence-categorisation/*SSU.fasta")        , optional: true, emit: ssu_fasta
    tuple val(meta), path("sequence-categorisation/*LSU.fasta")        , optional: true, emit: lsu_fasta
    tuple val(meta), path("sequence-categorisation/*5S.fasta")         , optional: true, emit: fiveS_fasta
    tuple val(meta), path("sequence-categorisation/*5_8S.fasta")       , optional: true, emit: five_eightS_fasta
    tuple val(meta), path("sequence-categorisation/*other_ncRNA.fasta"), optional: true, emit: ncrna_fasta
    tuple val(meta), path("*concat_SSU_LSU_coords.txt")                , emit: concat_ssu_lsu_coords
    path "versions.yml"                                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    get_subunits -i $easel_coords_fasta -n ${prefix}

    get_subunits_coords -i $matched_seqs_with_coords -s SSU -l LSU

    cat SSU_coords LSU_coords > ${prefix}_concat_SSU_LSU_coords.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        extractcoords: 0.1.2
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_concat_SSU_LSU_coords.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        extractcoords: 0.1.2
    END_VERSIONS
    """
}
