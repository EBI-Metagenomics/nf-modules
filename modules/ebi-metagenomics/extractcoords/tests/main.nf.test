
nextflow_process {

    name "Test Process EXTRACTCOORDS"
    script "../main.nf"
    process "EXTRACTCOORDS"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "extractcoords"

    test("extractcoords - should take a fasta file and an easel coordinates file and extract two subunit fasta files for the SSU and LSU, and a concatenated SSU and LSU coordinates file") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test_matched_seqs_with_coords.txt", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("extractcoords - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.fasta", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test_matched_seqs_with_coords.txt", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
