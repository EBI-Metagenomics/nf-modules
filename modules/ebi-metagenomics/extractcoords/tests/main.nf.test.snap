{
    "extractcoords - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ],
                "concat_ssu_lsu_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fiveS_fasta": [
                    
                ],
                "five_eightS_fasta": [
                    
                ],
                "lsu_fasta": [
                    
                ],
                "ncrna_fasta": [
                    
                ],
                "ssu_fasta": [
                    
                ],
                "versions": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-21T19:06:25.914889"
    },
    "extractcoords - should take a fasta file and an easel coordinates file and extract two subunit fasta files for the SSU and LSU, and a concatenated SSU and LSU coordinates file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,401222d263116bb145c1a67e6acc4709"
                    ]
                ],
                "6": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ],
                "concat_ssu_lsu_coords": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_concat_SSU_LSU_coords.txt:md5,401222d263116bb145c1a67e6acc4709"
                    ]
                ],
                "fiveS_fasta": [
                    
                ],
                "five_eightS_fasta": [
                    
                ],
                "lsu_fasta": [
                    
                ],
                "ncrna_fasta": [
                    
                ],
                "ssu_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_SSU.fasta:md5,fb23f13526672a300845b767dbc5fd44"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bb41e1add63c9b89d054972d8f663caf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-21T19:04:44.828471"
    }
}